milli/src/vector/mod.rs

```diff
@@ -26,6 +30,171 @@ pub type Embedding = Vec<f32>;
 
 pub const REQUEST_PARALLELISM: usize = 40;
 
+pub struct ArroyWrapper {
+    quantized: bool,
+    index: u16,
+    database: arroy::Database<Unspecified>,
+}
+
+impl ArroyWrapper {
+    pub fn new(database: arroy::Database<Unspecified>, index: u16, quantized: bool) -> Self {
+        Self { database, index, quantized }
+    }
+
+    pub fn index(&self) -> u16 {
+        self.index
+    }
+
+    pub fn dimensions(&self, rtxn: &RoTxn) -> Result<usize, arroy::Error> {
```

Comment:
Will this work on an empty db?

Reply:
If the DB has never been built, this’ll throw the missing metadata error.
If it has been built but contains no vectors, it should work

Reply:
will the db be built by meilisearch if it contains no vector?

Reply:
No